blackbox/docs/config/cluster.rst

```diff
@@ -309,74 +309,63 @@ resources. The following settings change the behaviour of those queries.
-  | *Default:*   ``3s``
-  | *Runtime:*  ``yes``
-
-  The time to wait for ping responses from other nodes when discovering.
-  Set this option to a higher value on a slow or congested network to minimize
-  discovery failures.
-
-.. _discovery.zen.ping_retries:
-
-**discovery.zen.ping_retries**
-  | *Default:*   ``3``
-  | *Runtime:*  ``yes``
-
-  How many ping failures (network timeouts) indicate that a node has failed.
-
-.. _discovery.zen.publish_timeout:
-
-**discovery.zen.publish_timeout**
-  | *Default:*   ``30s``
-  | *Runtime:*  ``yes``
+Data sharding and work splitting are at the core of CrateDB. This is how we
+manage to execute very fast queries over incredibly large datasets. In order
+for multiple CrateDB nodes to work together a cluster needs to be formed. The
+process of finding other nodes with which to form a cluster is called
+discovery. Discovery runs when a CrateDB node starts and when a node is not
+able to reach the master node and continues until a master node is found or a
+new master node is elected.
+
+.. _discovery_seed_hosts:
+
+**discovery.seed_hosts**
+   | *Default:* ``127.0.0.1``
+   | *Runtime:* ``no``
+
+   In order to form a cluster with CrateDB instances running on other nodes a
+   list of seed master-eligible nodes needs to be provided. This setting should
+   normally contain the addresses of all the master-eligible nodes in the
+   cluster. In order to seed the discovery process the nodes listed here must
+   be live and contactable. This setting contains either an array of hosts or a
+   comma-delimited string.
+   By default a node will bind to the available loopback and scan for local
+   ports between ``4300`` and ``4400`` to try to connect to other nodes running
+   on the same server. This default behaviour provides local auto clustering
+   without any configuration.
+   Each value should be in the form of host:port or host (where port defaults
+   to the setting ``transport.tcp.port``).
+ 
+.. NOTE::
 
-  Time a node is waiting for responses from other nodes to a published cluster
-  state.
+   IPv6 hosts must be bracketed.
 
-.. NOTE::
+**cluster.initial_master_nodes**
+   | *Default:* ``not set``
+   | *Runtime:* ``no``
 
-   Multicast used to be an option for node discovery, but was deprecated in
-   CrateDB 1.0.3 and removed in CrateDB 1.1.
+   Contains a list of names or IP addresses of the master-eligible nodes which
+   will vote in the very first election of a cluster that's bootstrapping for
+   the first time. By default this is not set, meaning it expects this node to
+   join an already formed cluster.
+   In development mode, with no discovery settings configured, this step is
+   performed by the nodes themselves, but this auto-bootstrapping is designed
+   to aim development and is not safe for production. In production you must
+   explicitly list the names or IP addresses of the master-eligible nodes whose
+   votes should be counted in the very first election.
```

Comment:
We should mention imho that this is only needed when a complete fresh cluster is started. For existing cluster full (re-)starts this is not needed.

Reply:
This is mentioned in the first paragraph "Contains a list of names or IP addresses of the master-eligible nodes which will vote in the very first election of a cluster that's bootstrapping for the first time." and also "whose votes should be counted in the very first election." Not sure how to make it more explicit. Do you have a suggestion here?

Reply:
You're right, I was reading too fast. Sounds already good, ignore my comment! ;-)